src/main/java/uk/gov/hmcts/darts/common/service/impl/AutomatedTaskServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    public String getAutomatedTaskCronExpression(AutomatedTask automatedTask) {
+        String cronExpression = getAutomatedTaskCronExpression(automatedTask.getTaskName());
+        if (cronExpression == null) {
+            log.info("Unable to find cron expression for task: {}", automatedTask.getTaskName());
+        }
+        return cronExpression;
+    }
+
+    public String getAutomatedTaskCronExpression(String taskName) {
+        String cronExpression = null;
+        Optional<AutomatedTaskEntity> automatedTaskEntity = getAutomatedTaskEntityByTaskName(taskName);
+        if (automatedTaskEntity.isPresent()) {
+            cronExpression = automatedTaskEntity.get().getCronExpression();
+            log.debug("{} cron expression: {}", taskName, cronExpression);
+        }
+        return cronExpression;
+    }
+
+    @Override
+    public void updateAutomatedTaskCronExpression(String taskName, String cronExpression) {
+        Optional<AutomatedTaskEntity> automatedTaskEntity = getAutomatedTaskEntityByTaskName(taskName);
+        if (automatedTaskEntity.isPresent()) {
+            AutomatedTaskEntity automatedTask = automatedTaskEntity.get();
+            if (CronExpression.isValidExpression(cronExpression)) {
+                automatedTask.setCronExpression(cronExpression);
+                automatedTaskRepository.saveAndFlush(automatedTask);
+                log.debug("Updated the task {} with cron expression {}", taskName, cronExpression);
+            }  else {
+                log.error("Unable to update the task {} with cron expression {}", taskName, cronExpression);
+                throw new DartsApiException(INVALID_CRON_EXPRESSION);
+            }
+        }  else {
+            log.error("Unable to update the task {} with cron expression {}", taskName, cronExpression);
+            throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+    }
+
+    @Override
+    public void loadTaskByName(String taskName) {
+        switch (taskName) {
+            case ProcessDailyListAutomatedTask.PROCESS_DAILY_LIST_TASK_NAME -> loadProcessDailyListAutomatedTask();
+            default -> throw new DartsApiException(FAILED_TO_FIND_AUTOMATED_TASK);
+        }
+    }
+
+    public void loadProcessDailyListAutomatedTask() {
+        ProcessDailyListAutomatedTask processDailyListAutomatedTask = new ProcessDailyListAutomatedTask();
+        Trigger trigger = createAutomatedTaskTrigger(processDailyListAutomatedTask);
+        taskScheduler.schedule(processDailyListAutomatedTask, trigger);
+    }
+
+    private void loadProcessDailyListAutomatedTask(ScheduledTaskRegistrar taskRegistrar) {
+        ProcessDailyListAutomatedTask processDailyListAutomatedTask = new ProcessDailyListAutomatedTask();
+        Trigger trigger = createAutomatedTaskTrigger(processDailyListAutomatedTask);
+        taskRegistrar.addTriggerTask(processDailyListAutomatedTask, trigger);
+    }
```

Comment:
It appears there are two different ways to load a scheduled task - via the `taskScheduler` or via the `taskRegister`. Are two ways necessary, or can we inify on a single approach?

Reply:
The taskRegister is only available by the configuration and is what contains the list of tasks even if they haven't run yet. The task scheduler which is available to the service can then start and stop tasks that have been loaded by the taskRegister. If you add just to task scheduler, you won't be able to amend it until has run for the first time.